vendors/nordic/nRF5_SDK_15.2.0/external/freertos/source/queue.c

```diff
@@ -1919,7 +1922,11 @@ Queue_t *pxQueue;
 
 	taskENTER_CRITICAL();
 	{
-		uxReturn = pxQueue->uxLength - pxQueue->uxMessagesWaiting;
+		if(pxQueue->uxMessagesWaiting <= pxQueue->uxLength){
+			uxReturn = pxQueue->uxLength - pxQueue->uxMessagesWaiting;
+		}else{
+			uxReturn = 0;
```

Comment:
I'm not sure about this. It seems to me that `pxQueue->uxMessagesWaiting <= pxQueue->uxLength` is an invariant, so this on an invariant violation branch. Silently returning doesn't seem like the best idea.

Deferring to @dcgaws @dan4thewin for logging or error handling ideas, but alternatively, I suggest `configASSERT( pxQueue->uxMessagesWaiting <= pxQueue->uxLength )` and then just do the subtraction.

Reply:
I think it is reasonable to return 0 for space available when the inequality fails.  I don't think that is failing silently.